File: algorithms/1394. Find Lucky Integer in an Array/solution.py
class Solution(object):
    def findLucky(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        return max(k if k == v else -1 for k, v in collections.Counter(arr).iteritems())
# Runtime: 48 ms, faster than 59.57% of Python online submissions for Find Lucky Integer in an Array.
# Memory Usage: 12.7 MB, less than 100.00% of Python online submissions for Find Lucky Integer in an Array.

File: algorithms/1394. Find Lucky Integer in an Array/solution.cc
class Solution {
public:
    int findLucky(vector<int>& arr) {
        std::unordered_map<int, int> m;
        for (int n : arr)
            m[n]++;
        int result = -1;
        for (auto it : m)
            result = it.first == it.second ? max(result, it.first) : result;
        return result;
    }
};

// Runtime: 12 ms, faster than 78.22% of C++ online submissions for Find Lucky Integer in an Array.
// Memory Usage: 10.4 MB, less than 100.00% of C++ online submissions for Find Lucky Integer in an Array.
